docs/content/development/build.md

```diff
@@ -2,28 +2,70 @@
+You can build Druid directly from source. Please note that these instructions are for building the latest stable version of Druid.
 For building the latest code in master, follow the instructions [here](https://github.com/druid-io/druid/blob/master/docs/content/development/build.md).
 
+# Requirements
+
 Building Druid requires the following:
-- [JDK 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
-- [Maven version 3.x](http://maven.apache.org/download.cgi)
 
-To do so, run these commands:
+- [JDK 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)+
+- [Maven 3.x](http://maven.apache.org/download.cgi)+
+
+# Building
+
+## Build a runtime distribution
+
+Move to the project root directory and run:
+
+```
+mvn clean package -DskipTests
+```
+
+This will compile the project and create the Druid binary distribution tarball under `distribution/target/druid-VERSION-bin.tar.gz`
+and `mysql-metadata-storage` extension under `distribution/target/mysql-metadata-storage-bin.tar.gz`.
+
+- For more information on `mysql-metadata-storage` extension, see [here](../development/extensions-core/mysql.html).
+- For more information on including extensions, see [here](../operations/including-extensions.html).
+
+## Build javadoc
+
+```
+mvn install -DskipTests && mvn javadoc:javadoc
+```
+
+# Testing
+
+## Run all unit/integration tests
 
 ```
-git clone git@github.com:druid-io/druid.git
-cd druid
-mvn clean package
+export AWS_REGION=us-east-1 && mvn test -B -Pparallel-test -Dmaven.fork.count=2
 ```
 
-This will compile the project and create the Druid binary distribution tar under
-`distribution/target/druid-VERSION-bin.tar.gz`.
+## Run specific module unit tests
+
+```
+mvn test -pl processing -B -Pparallel-test -Dmaven.fork.count=2 # run unit tests in processing module
+```
+
+## Run s3-extensions module unit tests
+
+```
+export AWS_REGION=us-east-1 && mvn test -pl s3-extensions -B -Pparallel-test -Dmaven.fork.count=2
+```
+
+## Run specific test suites
+
+```
+mvn test -pl common -pl sql -Dtest=BlockingPoolTest,ExpressionsTest
+```
+
+## Run specific test methods
+
+```
+mvn test -pl common -Dtest=BlockingPoolTest#testTake+testTakeTimeout
+```
 
-This will also create a tarball that contains `mysql-metadata-storage` extension under 
-`distribution/target/mysql-metadata-storage-bin.tar.gz`. If you want Druid to load `mysql-metadata-storage`, you can 
-first untar `druid-VERSION-bin.tar.gz`, then go to ```druid-<version>/extensions```, untar `mysql-metadata-storage-bin.tar.gz` 
-there. Now just specifiy `mysql-metadata-storage` in `druid.extensions.loadList` so that Druid will pick it up. 
-See [Including Extensions](../operations/including-extensions.html) for more information.
+Note: Druid uses considerable memory running its unit/integration tests. 8gb of memory is enough in general, but in
```

Comment:
I think we should check how much memory our tests use before adding this kind of caveat to give users more appropriate guidance without scaring them unnecessarily.